scrapegraphai/graphs/abstract_graph.py

```diff
@@ -65,6 +65,9 @@ def __init__(self, prompt: str, config: dict,
         self.cache_path = self.config.get("cache_path", False)
         self.browser_base = self.config.get("browser_base")
 
+        self.max_depth = config.get("max_depth", 3)  # Set max_depth from config, default to 3
```

Comment:
set `max_depth` default to 2 instead of 3, as we had evidence that on average it is the sweet spot; also place it in some common config file, you have that 3 running around in multiple files